lib/schema-registry.js

```diff
@@ -424,16 +415,10 @@ SchemaRegistry.prototype._registerSchemaLatest = Promise.method(function (schema
 
   log.debug('_registerSchemaLatest() :: Registered schema:', schemaObj.topic);
 
-  if (schemaObj.schemaType.toLowerCase() === 'value') {
-    this.schemaTypeById[`schema-${schemaObj.responseRaw.id}`] = schemaObj.type;
-    this.valueSchemas[schemaObj.topic] = schemaObj.type;
-    this.schemaIds[`${schemaObj.topic}-value`] = schemaObj.responseRaw.id;
-  } else {
-    this.schemaTypeById[`schema-${schemaObj.responseRaw.id}`] = schemaObj.type;
-    this.keySchemas[schemaObj.topic] = schemaObj.type;
-    this.schemaIds[`${schemaObj.topic}-key`] = schemaObj.responseRaw.id;
-  }
-  this.schemaMeta[schemaObj.topic] = schemaObj.responseRaw;
+  this.schemaTypeById[`schema-${schemaObj.responseRaw.id}`] = schemaObj.type;
+  this.schemas[schemaObj.schemaTopicRaw] = schemaObj.type;
+  this.schemaIds[schemaObj.schemaTopicRaw] = schemaObj.responseRaw.id;
+  this.schemaMeta[schemaObj.schemaTopicRaw] = schemaObj.responseRaw;
```

Comment:
Removing this.valueSchemas and this.keySchemas in favor of single schema container this.schemas